File: leetcode/algorithms/array/pascals-triangle.ts
function generate(numRows: number): number[][] {
    
    
    const arr: number[] = Array.from({
        length: numRows
    },(v,i) => i)
    
    return arr.reduce((a,i) => {
        a.push(i===0 ? [1] : Array.from({
            length: i+1
        },(v,j) => {
            return j===0 || j===i ? 1 : a[i-1][j-1]+a[i-1][j];
        }))
        return a;
    },new Array<Array<number>>())

};

File: leetcode/algorithms/array/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        List<List<Integer>> pascalList = new ArrayList<List<Integer>>();
        
        for(int i = 0;i<numRows;++i){
            if(i==0){
                pascalList.add(new ArrayList<Integer>(Arrays.asList(1)));
                continue;
            }
            List<Integer> currList = new ArrayList<Integer>();
            List<Integer> prevList = pascalList.get(i-1);
            // System.out.println(prevList.size());
            for(int j=0;j<=i;++j){
                currList.add(j==0 ? prevList.get(j) : prevList.get(j-1)+(j<prevList.size() ? prevList.get(j) : 0));
            }
            pascalList.add(currList);
        }
        
        return pascalList;
        
        
    }
}